src/daemon/control.rs

```diff
@@ -614,784 +604,35 @@ fn announce_spend_transaction(
-                    &revaultd.secp_ctx,
-                    xpub_ctx,
-                    &spend_tx.psbt,
-                    revaultd.managers_pubkeys.clone(),
-                    &spent_vaults,
-                ) {
-                    Ok(()) => {}
-                    Err(e) => {
-                        response_tx.send(Err(RpcControlError::SpendSignature(e.to_string())))?;
-                        continue;
-                    }
-                }
-
-                // Now we can ask all the cosigning servers for their signatures
-                log::debug!("Fetching signatures from Cosigning servers");
-                match fetch_cosigner_signatures(&revaultd, &mut spend_tx.psbt) {
-                    Ok(()) => {}
-                    Err(e) => {
-                        response_tx.send(Err(RpcControlError::Communication(e.to_string())))?;
-                        continue;
-                    }
-                }
-                let mut finalized_spend = spend_tx.psbt.clone();
-                match finalized_spend.finalize(&revaultd.secp_ctx) {
-                    Ok(()) => {}
-                    Err(e) => {
-                        response_tx.send(Err(RpcControlError::CosigningServer(format!(
-                            "Invalid signature given by the cosigners, psbt: '{}' (error: '{}')",
-                            spend_tx.psbt.as_psbt_string(),
-                            e
-                        ))))?;
-                        continue;
-                    }
-                }
-
-                // And then announce it to the Coordinator
-                let deposit_outpoints = spent_vaults
-                    .values()
-                    .map(|db_vault| db_vault.deposit_outpoint)
-                    .collect();
-                match announce_spend_transaction(&revaultd, finalized_spend, deposit_outpoints) {
-                    Ok(()) => {}
-                    Err(e) => {
-                        response_tx.send(Err(RpcControlError::Communication(e.to_string())))?;
-                        continue;
-                    }
-                }
-                db_update_spend(&db_path, &spend_tx.psbt)?;
-
-                // Finally we can broadcast the Unvault(s) transaction(s) and store the Spend
-                // transaction for later broadcast
-                match bitcoind_broadcast_unvaults(
-                    &bitcoind_tx,
-                    &db_path,
-                    &revaultd.secp_ctx,
-                    &spent_vaults,
-                ) {
-                    Ok(()) => {}
-                    Err(e) => {
-                        response_tx.send(Err(RpcControlError::UnvaultBroadcast(e.to_string())))?;
-                        continue;
-                    }
-                }
-                db_mark_broadcastable_spend(&db_path, &spend_txid)?;
+#[derive(Clone)]
+pub struct RpcUtils {
+    pub revaultd: Arc<RwLock<RevaultD>>,
+    pub db_path: PathBuf,
+    pub network: Network,
```

Comment:
These are a duplicate from RevaultD, i think you should be retrieving them from it instead